clang/lib/Sema/SemaDecl.cpp

```diff
@@ -3998,7 +3999,8 @@ bool Sema::MergeFunctionDecl(FunctionDecl *New, NamedDecl *&OldD, Scope *S,
       }
     }
 
-    if (Context.typesAreCompatible(OldQType, NewQType)) {
+    if (Context.typesAreCompatible(OldQType, NewQType) ||
+        getLangOpts().IgnoreConflictingTypes) {
```

Comment:
I don't fully understand which part this is checking, but just want to double-check if this is looser than we want—we don't want to end up letting people accidentally break their code much more while they have the workaround applied.

Reply:
The option itself now is a bit misleading since it doesn't entirely ignore the validation on this class of errors, but reverts back to the behavior on swift/release/5.7. I think that is what we want and is alright.
We really don't want people to be using this outside of what you specified already.